torch_xla/distributed/spmd/api.py

```diff
@@ -0,0 +1,196 @@
+import logging
+import os
+from functools import wraps
+from typing import Any, Callable, Dict, Optional, Sequence, Tuple, Union
+
+import torch
+
+import torch.nn as nn
+from torch.distributed._tensor.device_mesh import DeviceMesh
+from torch.distributed._tensor.placement_types import Placement, Replicate
+
+log = logging.getLogger(__name__)
+
+TORCH_XLA_INITIALIZED = False
+try:
+  import torch_xla.core.xla_model as xm  # type:ignore[import]  # noqa: F401
+  import torch_xla.runtime as xr  # type:ignore[import]
+  from torch_xla.experimental.spmd import (  # type:ignore[import]
+      XLAShardedTensor,)
+  from torch_xla.experimental.spmd import (  # type:ignore[import]
+      mark_sharding, Mesh, ShardingType,
+  )
+
+  TORCH_XLA_INITIALIZED = True
+except ImportError as e:
+  log.warning(e.msg)
+
+
+# wrapper to check xla test requirements
+def with_xla(func: Callable) -> Callable:
+  assert func is not None
+
+  @wraps(func)  # pyre-ignore[6]
+  def wrapper(
+      self,
+      *args: Tuple[object],
+      **kwargs: Dict[str, Any]  # type: ignore[misc]
+  ) -> None:
+    if TORCH_XLA_INITIALIZED:
+      # TODO(yeounoh) replace this with xr.use_spmd() when we deprecate the flag.
+      os.environ["XLA_USE_SPMD"] = "1"
```

Comment:
why not just call `xr.use_spmd()` now?

Reply:
there could be a case where the caller already has initialized some tensors before entering the API. We may want to support both, at least in the implementation. I will leave as a TODO here.